File: 36-valid-sudoku/36-valid-sudoku.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
        rows = collections.defaultdict(set)
        cols = collections.defaultdict(set)
        sqs = collections.defaultdict(set)
        
        for r in range(9):
            for c in range(9):
                if board[r][c] == '.' or board[r][c] == ',':
                    continue
                elif (board[r][c] in rows[r] or
                     board[r][c]  in cols[c] or
                     board[r][c]  in sqs[(r//3,c//3)]):
                    return False
                else:
                    rows[r].add(board[r][c])
                    cols[c].add(board[r][c])
                    sqs[(r//3,c//3)].add(board[r][c])
        return True

File: 36-valid-sudoku/36-valid-sudoku.java
class Solution {
    public boolean isValidSudoku(char[][] board) {
        HashSet<String> map = new HashSet<>();
        
        for(int r =0; r<9;r++){
            for(int c =0; c<9;c++){
                if (board[r][c] != '.'){
                    if (map.contains("row" + r + board[r][c]) || map.contains("col" + c + board[r][c])){
                        return false;
                    }
                    map.add("row" + r + board[r][c]);
                    map.add("col" + c + board[r][c]);
                    
                    if(map.contains("box" + (r/3) + (c/3) + board[r][c])){
                        return false;
                    }
                    map.add("box" + (r/3) + (c/3) + board[r][c]);
                }
            }
        }
        return true;
    }
}